File: 10-scope/Scope.java
public class Scope {

    public int x = 3;

    public void fun1() {
        int x = 4;
        fun2();
    }

    public void fun2() {
        System.out.println(x);
    }

    public static void main(String[] args) {
        Scope thingy = new Scope();
        thingy.fun1();
    }
}

File: 10-scope/scope.c
#include <stdio.h>

// Function prototypes so I don't have to worry about the order they appear in
void fun1();
void fun2();

int x = 3;

void fun1() {
    int x = 4;
    fun2();
}

void fun2() {
    printf("%i\n", x);
}

int main() {
    fun1();
}
